pyomo/contrib/doe/doe.py

```diff
@@ -1302,133 +1216,1015 @@ def det_general(m):
+        # an example filter: (self.store_all_results_dataframe["CA0"]==5).
+        if len(fixed_design_variables.keys()) != 0:
+            filter = ""
+            i = 0
+            for k, v in fixed_design_variables.items():
+                filter += "(results_pd['"
+                filter += str(k)
+                filter += "']=="
+                filter += str(v)
+                filter += ")"
+                if i < (len(fixed_design_variables.keys()) - 1):
+                    filter += "&"
+                i += 1
+            # extract results with other dimensions fixed
+            figure_result_data = results_pd.loc[eval(filter)]
+
+        # if there is no other fixed dimensions
+        else:
+            figure_result_data = results_pd
+
+        # Add attributes for drawing figures in later functions
+        self.figure_result_data = figure_result_data
+        self.figure_sens_des_vars = sensitivity_design_variables
+        self.figure_fixed_des_vars = fixed_design_variables
+
+        # if one design variable name is given as DOF, draw 1D sensitivity curve
+        if len(self.figure_sens_des_vars) == 1:
+            self._curve1D(
+                title_text,
+                xlabel_text,
+                font_axes=font_axes,
+                font_tick=font_tick,
+                log_scale=log_scale,
+                figure_file_name=figure_file_name,
+            )
+        # if two design variable names are given as DOF, draw 2D heatmaps
+        elif len(self.figure_sens_des_vars) == 2:
+            self._heatmap(
+                title_text,
+                xlabel_text,
+                ylabel_text,
+                font_axes=font_axes,
+                font_tick=font_tick,
+                log_scale=log_scale,
+                figure_file_name=figure_file_name,
+            )
+        # ToDo: Add the multidimensional plotting
+        else:
+            pass
 
-    def _fix_design(self, m, design_val, fix_opt=True, optimize_option=None):
+    def _curve1D(
+        self,
+        title_text,
+        xlabel_text,
+        font_axes=16,
+        font_tick=14,
+        figure_file_name=None,
+        log_scale=True,
+    ):
         """
-        Fix design variable
+        Draw 1D sensitivity curves for all design criteria
 
         Parameters
         ----------
-        m: model
-        design_val: design variable values dict
-        fix_opt: if True, fix. Else, unfix
-        optimize: a dictionary, keys are design variable name, values are True or False, deciding if this design variable is optimized as DOF this time
+        title_text: name of the figure, a string
+        xlabel_text: x label title, a string.
+            In a 1D sensitivity curve, it is the design variable by which the curve is drawn.
+        font_axes: axes label font size
+        font_tick: tick label font size
+        log_scale: if True, the result matrix will be scaled by log10
```

Comment:
The docstring doesn't match the implementation. `figure_file_name` is missing from the docstring.